lib/cache.js

```diff
@@ -0,0 +1,107 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict'
+
+const crypto = require('crypto')
+const Memcached = require('memcached')
+const P = require('./promise')
+
+P.promisifyAll(Memcached.prototype)
+
+const NOP = () => P.resolve()
+const NULL_CACHE = {
+  delAsync: NOP,
+  getAsync: NOP,
+  setAsync: NOP
+}
+
+module.exports = (log, config, namespace) => {
+  let _cache
+
+  const CACHE_ADDRESS = config.memcached.address
+  const CACHE_IDLE = config.memcached.idle
+  const CACHE_LIFETIME = config.memcached.lifetime
+
+  return {
+    /**
+     * Delete data from the cache, keyed by a hash of
+     * `token.uid` and `token.id`. Fails silently if
+     * the cache is not enabled.
+     *
+     * @param token
+     */
+    del (token) {
```

Comment:
Ah gotcha, generic came off to me as any key and data, which I thought was misleading because this requires a token of sorts to use. Maybe `token_cache` is a better name?